assets/app/view/game/buy_trains.rb

```diff
@@ -19,7 +19,7 @@ def render_president_contributions
         player = @corporation.owner
         children = []
 
-        verb = @must_buy_train ? 'must' : 'may'
+        verb = @must_buy_train || @step.must_buy_train_if_able_to?(@corporation) ? 'must' : 'may'
```

Comment:
why is this needed

Reply:
1877: Stockholm Tramways only requires corporations to buy a train if it's possible to buy the cheapest train from the depot (using president's contribution). Thus `@must_buy_train` may be `false` while I still want to tell the player how much he "must" contribute to buy the cheapest train from the depot.

However, I realize now that the current code is wrong on one point, if a corporation decides not to buy a train when it has none, the president still has to contribute as much as he can, something which is not reflected in the code or in the text prompt, i.e. this code has to be fixed either way before merging. What I need instead is to make an option for this function to output a prompt that tells the player he must either buy a train or contribute all his cash to the company.